jax/interpreters/partial_eval.py

```diff
@@ -998,13 +999,18 @@ def call_partial_eval_custom_rule(
   # by convention, _partial_eval_jaxpr_custom drops units on known outputs
   known_units_out = [v.aval is core.abstract_unit for v in jaxpr.outvars]
   dropped_outs_known = map(op.or_, unks_out, known_units_out)
+  kept_outs_known = [not d for d in dropped_outs_known]
   out_binders_known, _ = partition_list(dropped_outs_known, eqn.outvars)
   _, out_binders_staged = partition_list(inst_out, eqn.outvars)
+  kept_outs_staged = inst_out
   newvar = core.gensym([jaxpr_known, jaxpr_staged])
   residuals = [newvar(v.aval) for v in jaxpr_staged.invars[:num_res]]
-  params_known = dict(eqn.params, call_jaxpr=jaxpr_known)
-  params_staged = dict(eqn.params, call_jaxpr=jaxpr_staged)
-  params_known, params_staged = params_updater(unks_in, num_res, params_known, params_staged)
+  params_known = dict(eqn.params)
+  params_known[jaxpr_param_name] = jaxpr_known
```

Comment:
```suggestion
  params_known = {jaxpr_param_name: jaxpr_known, **eqn.params}
```

⛳

Reply:
I tried this and it didn't work. But `{**eqn.params, jaxpr_param_name: jaxpr_known}` did!